configd/src/apps/su/main.cpp

```diff
@@ -28,6 +28,19 @@ int main(int argc, char** argv)
     gid_t g = p->pw_gid;
     uid_t u = p->pw_uid;
 
+    gid_t grouplist[256];
+    int group_arr_sz = 256;
+#ifdef __APPLE__
+    int mac_gid = g;
+    int mac_groups[256];
+    int ggl = getgrouplist(username, mac_gid, mac_groups, &group_arr_sz);
+    for (int i = 0; i < group_arr_sz; ++i) {
```

Comment:
This loop should only be called if `ggl == 0` since `getgrouplist` returns 0 on success and -1 on failure on darwin.
`mac_groups` should only be read If `getgrouplist` succeeded.